File: src/jvmMain/java/resaver/pex/VDataFlt.java
package resaver.pex;

import java.io.IOException;
import java.nio.ByteBuffer;

import static resaver.pex.DataType.FLOAT;

/**
 * VData that stores a float.
 */
public class VDataFlt extends VData {

    VDataFlt(float val) {
        this.VALUE = val;
    }

    @Override
    public void write(ByteBuffer output) throws IOException {
        output.put((byte) this.getType().ordinal());
        output.putFloat(this.VALUE);
    }

    @Override
    public int calculateSize() {
        return 5;
    }

    @Override
    public DataType getType() {
        return FLOAT;
    }

    @Override
    public String toString() {
        return String.format("%g", this.VALUE);
    }

    public float getValue() {
        return this.VALUE;
    }

    @Override
    public int hashCode() {
        int hash = 7;
        hash = 83 * hash + Float.hashCode(this.VALUE);
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        } else if (obj == null) {
            return false;
        } else if (getClass() != obj.getClass()) {
            return false;
        }
        final VDataFlt other = (VDataFlt) obj;
        return this.VALUE == other.VALUE;
    }

    final private float VALUE;
}

File: src/jvmMain/kotlin/resaver/pex/VDataFlt.kt
package resaver.pex

import PlatformByteBuffer
import java.io.IOException

/**
 * VData that stores a float.
 */
class VDataFlt internal constructor(val value: Float) : VData() {
    @Throws(IOException::class)
    override fun write(output: PlatformByteBuffer?) {
        output?.put(type.ordinal.toByte())
        output?.putFloat(value)
    }

    override fun calculateSize(): Int {
        return 5
    }

    override val type: DataType
        get() = DataType.FLOAT

    override fun toString(): String {
        return String.format("%g", value)
    }

    override fun hashCode(): Int {
        var hash = 7
        hash = 83 * hash + value.hashCode()
        return hash
    }

    override fun equals(other: Any?): Boolean {
        return when {
            this === other -> {
                true
            }
            other == null -> {
                false
            }
            javaClass != other.javaClass -> {
                false
            }
            else -> {
                val other2 = other as VDataFlt
                value == other2.value
            }
        }
    }

}